### DISCUSSION TITLE
Input field value do not switch dynamically on condition

### DISCUSSION THREAD

**Discussion by kosaki08** [react-hook-form/react-hook-form#9923] [Q&A] [answered]
### Version Number
7.42.1

### Codesandbox/Expo snack

https://codesandbox.io/s/laughing-fog-s2l3vs?file=/src/App.tsx

### Steps to reproduce

1. Select Answer by radio button.
2. (The value in the Answer input is the same as the Question value.)

### Expected behavior

If we select Answer, the Answer's value (blank in this) is displayed, and the value for Question is set by switching the radio button in the same way.

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

No response

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@kosaki08 

Please always add prop `key` on the target element type that is switching field name. Adding prop `key` on wrapper is no help as the wrapper is not re-rendering in this case.

Working demo: https://codesandbox.io/s/serene-cache-yw6uen?file=/src/TestForm.tsx

**Reply by kosaki08** [reply]
@leapful 
Thank you. I understand the issue with my code.
